packages/voter-dapp/run_tests.sh

```diff
@@ -34,6 +34,17 @@ echo '#                                                                  #'
 echo '# 1/16. Deploying voting contracts on test network                 #'
 echo '#                                                                  #'
 echo -e '####################################################################\n'
+
+# This hacky snippet just copies the newer voting bytecode into the package artifacts that the voter dapp uses.
+# This forces the voter dapp to deploy the _new_ voting contract with all the other contracts unchanged, which will
+# force it to test the old 2key contract with the new voting contract.
```

Comment:
This is a well done hack and I really like how you backup the old ABI. Perhaps a more stable fix longer term to this test would allow user to enter in contract versions that they want to test (for 2key and voting) and perform backups where appropriate.

Reply:
Agreed, that would be awesome. When we do a full dapp upgrade to the new contracts, we should roll this hacky change back in its entirety. I don't want this code to last more than a week :)